services/workflows-service/src/data-analytics/data-analytics.service.ts

```diff
@@ -170,7 +163,44 @@ export class DataAnalyticsService {
         query = Prisma.sql`SELECT ${selectClause}, COUNT(id) AS "transactionCount" FROM "TransactionRecord" tr WHERE ${whereClause} GROUP BY ${groupByClause}`;
     }
 
-    return await this._executeQuery(query);
+    return await this._executeQuery<Array<Record<string, unknown>>>(query);
+  }
+
+  async evaluateHighTransactionTypePercentage({
+    projectId,
+    transactionType,
+    subjectColumn,
+    minimumCount,
+    minimumPercentage,
+    timeAmount,
+    timeUnit,
+  }: HighTransactionTypePercentage) {
+    return await this._executeQuery<Array<{ counterpartyId: string }>>(Prisma.sql`
+      WITH "transactionsData" AS (
+        SELECT
+          "${Prisma.raw(subjectColumn)}",
+          COUNT(*) AS "transactionCount",
+          COUNT(*) FILTER (WHERE "transactionType" = '${Prisma.raw(
+            transactionType,
+          )}') AS "filteredTransactionCount"
```

Comment:
can you use `Prisma.sql` which handle parameter injection in here

Reply:
It doesn't work with `Prisma.sql` because it's an enum or something. We already use `Prisma.raw` in other eval functions in this file.

Reply:
it works with enum in other places, we should minimize prisma.raw as much as we can

Reply:
```ts
    const conditions: Prisma.Sql[] = [
      Prisma.sql`tr."projectId" = '${projectId}'`,
      Prisma.sql`tr."businessId" IS NOT NULL`,
      // TODO: should we use equation instead of IN clause?
      Prisma.sql`tr."transactionType"::text IN (${Prisma.join(transactionType, ',')})`,
      Prisma.sql`"transactionDate" >= CURRENT_DATE - INTERVAL '${Prisma.raw(
        `${timeAmount} ${timeUnit}`,
      )}'`,
    ];
```